### PR TITLE
Do not show queues with deleted vhost.

### LINKED ISSUE #1300: `down` durable queues are not deleted when deleting the vhost.
Durable queues, which are considered `down` (e.g. host node is not running), are not cleaned up when deleting the vhost.
Steps to reproduce:
1. start 2-nodes cluster
2. create a queue on node 2
3. stop node 2 (queue will be displayed as `down` in mgmt and list_queues)
4. delete the vhost for the queue 
5. `rabbitmqctl list_queues -p <vhost>` will display the queue.

When node 2 is restarted, the queue is started again even though vhost is deleted. 

This happens because deleting a vhost deletes all the queues in `rabbit_amqqueue:list/1`, which does not include `down` queues.

### PR DESCRIPTION
**Pull request by hairyhum**
When vhost does not exist for a queue, which reside on
a stopped node, the queue will not be displayed in the 'list_queues'
command output.
When a node goes back up, queues without vhost will be deleted.
[Fixes #1300]
[#149292743]

### CODE REVIEW

File: src/rabbit_amqqueue.erl [hunk 1]
```diff
@@ -257,6 +258,31 @@ start(Qs) ->
...
+        DurableQueues),
+
+    {LocalQueuesWithoutVhost, _RemoteQueuesWithoutVhost} = lists:partition(
+        fun(#amqqueue{pid = QPid}) -> node(QPid) == node() end,
+        QueuesWithoutVhost),
+
+    {atomic, ok} =
+        mnesia:sync_transaction(
+            fun () ->
+                rabbit_log:error("Deleting ~p~n", [LocalQueuesWithoutVhost]),
```
Comment (lukebakken) [inline src/rabbit_amqqueue.erl]: Should this be `info` rather than `error`?

Comment (michaelklishin) [inline src/rabbit_amqqueue.erl]: I'd make it a warning and explain why it's being deleted.

Comment (hairyhum) [inline src/rabbit_amqqueue.erl]: That was a debug message. Do you think it should stay?

**Comment by michaelklishin** [comment]
Doh, I hit Merge in the wrong tab. Sorry.

**Comment by michaelklishin** [comment]
We agreed to do a couple of small adjustments in `stable`. The PR looks good in general.

**Pull request merged**